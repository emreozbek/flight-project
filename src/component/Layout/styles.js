export const styles = theme => ({
  background: {
    display: "block",
    content: "",
    width: "100%",
    height: theme.spacing.unit * 30,
    backgroundColor: theme.palette.secondary.main,
    position: "absolute",
    zIndex: -1
  },
  wrapper: {
    padding: `${theme.spacing.unit * 8}px ${theme.spacing.unit * 2}px 0 ${theme
      .spacing.unit * 2}px`,
    maxWidth: 720 + theme.spacing.unit * 4,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
