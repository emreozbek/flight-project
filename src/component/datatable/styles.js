export const styles = theme => ({
  table: {
    minWidth: 720
  },
  tableWrapper: {
    overflowX: "auto"
  },
  searchBox: {
    paddingTop: theme.spacing.unit * 2,
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  }
});
