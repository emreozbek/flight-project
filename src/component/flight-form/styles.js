export const styles = theme => ({
  padding: {
    padding: theme.spacing.unit * 2
  },
  marginLeft: {
    marginLeft: theme.spacing.unit * 2
  }
});
