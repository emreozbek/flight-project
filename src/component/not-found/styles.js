export const styles = theme => ({
  padding: {
    padding: theme.spacing.unit * 2
  }
});
