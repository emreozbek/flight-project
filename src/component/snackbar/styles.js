export const styles = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  },
  error: {
    backgroundColor: theme.palette.error.main
  }
});
