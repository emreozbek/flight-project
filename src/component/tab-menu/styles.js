export const styles = theme => ({
  content: {
    position: "relative",
    minHeight: 256
  }
});
