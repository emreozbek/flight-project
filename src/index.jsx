import React from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { Router } from "react-router";
import App from "./component/App";
import { store, history } from "./store";

render(
  <Provider store={store}>
    <Router history={history}>
      <App />
    </Router>
  </Provider>,
  document.getElementById("app")
);
