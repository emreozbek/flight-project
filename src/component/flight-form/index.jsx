import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Field, reduxForm } from "redux-form";
import { Grid, Button, withStyles } from "@material-ui/core";
import moment from "moment";
import { addFlight } from "../../action";
import {
  renderTextField,
  renderRadioGroup,
  renderDateTimePicker
} from "../../utility/field-renderer";
import { styles } from "./styles";
import { DATE_FORMAT, MESSAGE } from "../../config";
import { validate } from "./validate";

export class FlightForm extends Component {
  render() {
    const { classes, addFlight, reset, handleSubmit } = this.props;
    return (
      <form
        onSubmit={handleSubmit(values => {
          values.departureDate = moment(values.departureDate).format(
            DATE_FORMAT
          );
          values.arrivalDate = moment(values.arrivalDate).format(DATE_FORMAT);
          addFlight(values);
          reset();
        })}
        className={classes.padding}
        noValidate="novalidate"
      >
        <Grid container spacing={16}>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              required
              helperText={MESSAGE.form.required}
              name="departure"
              component={renderTextField}
              label="Departure"
              margin="dense"
              fullWidth
            />
          </Grid>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              required
              name="arrival"
              helperText={MESSAGE.form.required}
              component={renderTextField}
              label="Arrival"
              margin="dense"
              fullWidth
            />
          </Grid>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              name="departureDate"
              component={renderDateTimePicker}
              helperText={MESSAGE.form.required}
              label="Departure Date"
              margin="dense"
              fullWidth
              required
            />
          </Grid>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              name="arrivalDate"
              component={renderDateTimePicker}
              helperText={MESSAGE.form.required}
              label="Arrival Date"
              margin="dense"
              fullWidth
              required
            />
          </Grid>
          <Grid item xs={12} sm={12} className={classes.padding}>
            <Field
              name="flightType"
              label="Flight Type"
              component={renderRadioGroup}
              options={[
                { value: "Cheap", label: "Cheap" },
                { value: "Business", label: "Business" }
              ]}
              row
            />
          </Grid>
        </Grid>
        <Button variant="outlined" color="primary" type="submit">
          Save
        </Button>
        <Button
          color="secondary"
          className={classes.marginLeft}
          onClick={() => {
            reset();
          }}
        >
          Reset
        </Button>
      </form>
    );
  }
}
FlightForm.propTypes = {
  classes: PropTypes.object,
  addFlight: PropTypes.func,
  handleSubmit: PropTypes.func
};
const flightForm = reduxForm({
  form: "flight",
  destroyOnUnmount: false,
  validate
})(FlightForm);
const mapDispatchToProps = {
  addFlight
};
export default connect(
  null,
  mapDispatchToProps
)(withStyles(styles)(flightForm));
