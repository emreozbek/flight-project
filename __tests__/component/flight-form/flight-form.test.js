import React from "react";
import renderer from "react-test-renderer";
import { reduxForm } from "redux-form";
import { Provider } from "react-redux";
import { withStyles } from "@material-ui/core";
import { FlightForm } from "../../../src/component/flight-form";
import { styles } from "../../../src/component/flight-form/styles";
import { store } from "../../../src/store";

const Decorated = withStyles(styles)(
  reduxForm({ form: "flightForm" })(FlightForm)
);
describe("Flight Form", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Decorated />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
