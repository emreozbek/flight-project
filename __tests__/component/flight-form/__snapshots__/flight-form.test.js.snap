// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Form renders correctly 1`] = `
<form
  className="ReduxForm-padding-1"
  noValidate="novalidate"
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-container-3 MuiGrid-spacing-xs-16-26"
  >
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-6-51 ReduxForm-padding-1"
    >
      <div
        className="MuiFormControl-root-100 MuiFormControl-marginDense-102 MuiFormControl-fullWidth-103"
        onDragStart={[Function]}
        onDrop={[Function]}
      >
        <label
          className="MuiFormLabel-root-115 MuiFormLabel-required-120 MuiInputLabel-required-108 MuiInputLabel-root-104 MuiInputLabel-formControl-109 MuiInputLabel-animated-112 MuiInputLabel-marginDense-110"
          data-shrink={false}
        >
          Departure
          <span
            className="MuiFormLabel-asterisk-121"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-135 MuiInput-root-122 MuiInput-underline-126 MuiInputBase-fullWidth-144 MuiInput-fullWidth-129 MuiInputBase-formControl-136 MuiInput-formControl-123 MuiInputBase-marginDense-142"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-145 MuiInput-input-130 MuiInputBase-inputMarginDense-146 MuiInput-inputMarginDense-131"
            disabled={false}
            name="departure"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-6-51 ReduxForm-padding-1"
    >
      <div
        className="MuiFormControl-root-100 MuiFormControl-marginDense-102 MuiFormControl-fullWidth-103"
        onDragStart={[Function]}
        onDrop={[Function]}
      >
        <label
          className="MuiFormLabel-root-115 MuiFormLabel-required-120 MuiInputLabel-required-108 MuiInputLabel-root-104 MuiInputLabel-formControl-109 MuiInputLabel-animated-112 MuiInputLabel-marginDense-110"
          data-shrink={false}
        >
          Arrival
          <span
            className="MuiFormLabel-asterisk-121"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-135 MuiInput-root-122 MuiInput-underline-126 MuiInputBase-fullWidth-144 MuiInput-fullWidth-129 MuiInputBase-formControl-136 MuiInput-formControl-123 MuiInputBase-marginDense-142"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-145 MuiInput-input-130 MuiInputBase-inputMarginDense-146 MuiInput-inputMarginDense-131"
            disabled={false}
            name="arrival"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-6-51 ReduxForm-padding-1"
    >
      <div
        className="MuiFormControl-root-100 MuiFormControl-marginDense-102 MuiFormControl-fullWidth-103"
        onClick={[Function]}
        onKeyPress={[Function]}
      >
        <label
          className="MuiFormLabel-root-115 MuiFormLabel-required-120 MuiInputLabel-required-108 MuiInputLabel-root-104 MuiInputLabel-formControl-109 MuiInputLabel-animated-112 MuiInputLabel-marginDense-110"
          data-shrink={false}
        >
          Departure Date
          <span
            className="MuiFormLabel-asterisk-121"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-135 MuiInput-root-122 MuiInput-underline-126 MuiInputBase-fullWidth-144 MuiInput-fullWidth-129 MuiInputBase-formControl-136 MuiInput-formControl-123 MuiInputBase-marginDense-142"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-145 MuiInput-input-130 MuiInputBase-inputMarginDense-146 MuiInput-inputMarginDense-131"
            disabled={false}
            mask={null}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            pipe={null}
            readOnly={true}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-6-51 ReduxForm-padding-1"
    >
      <div
        className="MuiFormControl-root-100 MuiFormControl-marginDense-102 MuiFormControl-fullWidth-103"
        onClick={[Function]}
        onKeyPress={[Function]}
      >
        <label
          className="MuiFormLabel-root-115 MuiFormLabel-required-120 MuiInputLabel-required-108 MuiInputLabel-root-104 MuiInputLabel-formControl-109 MuiInputLabel-animated-112 MuiInputLabel-marginDense-110"
          data-shrink={false}
        >
          Arrival Date
          <span
            className="MuiFormLabel-asterisk-121"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-135 MuiInput-root-122 MuiInput-underline-126 MuiInputBase-fullWidth-144 MuiInput-fullWidth-129 MuiInputBase-formControl-136 MuiInput-formControl-123 MuiInputBase-marginDense-142"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-145 MuiInput-input-130 MuiInputBase-inputMarginDense-146 MuiInput-inputMarginDense-131"
            disabled={false}
            mask={null}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            pipe={null}
            readOnly={true}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-12-43 MuiGrid-grid-sm-12-57 ReduxForm-padding-1"
    >
      <fieldset
        className="MuiFormControl-root-100"
      >
        <legend
          className="MuiFormLabel-root-115 MuiFormLabel-required-120"
        >
          Flight Type
          <span
            className="MuiFormLabel-asterisk-121"
          >
             *
          </span>
        </legend>
        <div
          className="MuiFormGroup-root-176 MuiFormGroup-row-177"
          onBlur={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          role="radiogroup"
        >
          <label
            className="MuiFormControlLabel-root-178"
          >
            <span
              className="MuiButtonBase-root-199 MuiIconButton-root-193 MuiPrivateSwitchBase-root-189 MuiRadio-root-184 MuiRadio-colorSecondary-188"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-198"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-202"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-192"
                  disabled={false}
                  onChange={[Function]}
                  type="radio"
                  value="Cheap"
                />
              </span>
              <span
                className="MuiTouchRipple-root-273"
              />
            </span>
            <span
              className="MuiTypography-root-211 MuiTypography-body1-220 MuiFormControlLabel-label-183"
            >
              Cheap
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root-178"
          >
            <span
              className="MuiButtonBase-root-199 MuiIconButton-root-193 MuiPrivateSwitchBase-root-189 MuiRadio-root-184 MuiRadio-colorSecondary-188"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-198"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-202"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-192"
                  disabled={false}
                  onChange={[Function]}
                  type="radio"
                  value="Business"
                />
              </span>
              <span
                className="MuiTouchRipple-root-273"
              />
            </span>
            <span
              className="MuiTypography-root-211 MuiTypography-body1-220 MuiFormControlLabel-label-183"
            >
              Business
            </span>
          </label>
        </div>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root-199 MuiButton-root-247 MuiButton-outlined-255 MuiButton-outlinedPrimary-256"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-248"
    >
      Save
    </span>
    <span
      className="MuiTouchRipple-root-273"
    />
  </button>
  <button
    className="MuiButtonBase-root-199 MuiButton-root-247 MuiButton-text-249 MuiButton-textSecondary-251 MuiButton-flat-252 MuiButton-flatSecondary-254 ReduxForm-marginLeft-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-248"
    >
      Reset
    </span>
    <span
      className="MuiTouchRipple-root-273"
    />
  </button>
</form>
`;
